### DISCUSSION TITLE
Pasting in server

### DISCUSSION THREAD

**Discussion by Leopotter1218** [maruohon/litematica#535] [Q&A]
i have all necessary permissions but it is showing some errorcan u hep me resolve it
![2022-05-18_18 35 11](https://user-images.githubusercontent.com/105574745/169045388-f5566ad7-ba6e-4eac-bec6-77954cb7635c.png)

**Comment by maruohon** [comment]
Does the server have World Edit installed at all? Either it doesn't, or you don't have the permission to use it, I guess? Did you try using the vanilla commands instead of enabling the World Edit option? Or do you not have the permission for the `/setblock` and `/fill` commands either?

**Reply by Leopotter1218** [reply]
We don't have world edit installed but i have perms for all other commands including setblock and fill as i have used these commands before

**Reply by maruohon** [reply]
Then why did you enable the World Edit option in Litematica? Disable the Generic -> `commandUseWorldEdit` option if the server doesn't have World Edit.

**Reply by Leopotter1218** [reply]
can the delete option in litematica be undont  the one in tools
the pasting problem was solved thank you

**Reply by Leopotter1218** [reply]
why are some bloakcs being not pasted
![2022-05-19_18 29 11](https://user-images.githubusercontent.com/105574745/169299171-7846a6ac-296f-4cc5-bd6e-bb180e28ba31.png)

**Comment by maruohon** [comment]
There is no undo functionality yet. It will be added at some point, but it will basically just be automatic schematics being saved of the affected area, and that will have the same limitations in multiplayer as normal schematic saving, i.e. it won't be able to save block entity and entity data and scheduled block ticks, until the server-side support for Litematica gets added to the Servux mod, which would then obviously need to be installed on the server side for that stuff to work. In that case I think the undo support would likely even be handled via Servux directly on the server side somehow.

As to some block not being pasted... is there anything in those positions, and are you using replace behavior `None` or one of the other options like `All`? Or is it just Signs that are having issues?

**Reply by Leopotter1218** [reply]
The replace is set to all and deepslate tinted glass signs pistons comparators have the same issue

**Reply by Leopotter1218** [reply]
So then for the undo to be done if i install servux now can it be done?

**Reply by maruohon** [reply]
It doesn't exist yet in either mod. I mentioned that it's planned for it, but no idea when I'll have time to work on it.

As for the blocks not getting pasted... I don't really have an idea what would cause it. Maybe try disabling the `pasteUseFillCommand` option in Generic, if it's currently enabled. Does it always spam the "No block were filled" message about those? Does it work in any other location, or is it always the same blocks missing regardless of the paste location?

**Reply by Leopotter1218** [reply]
if i apply the command agin it says 423 blocks were filled using setblocks which are usually the blocks that are already filled I will try disabling the command and reply to next time

**Reply by Leopotter1218** [reply]
it does not work  it also did no paste some tintedglass does litematica require servux to work on servers?

**Reply by maruohon** [reply]
Currently Servux doesn't do anything at all for Litematica. In the 0.1.x version it only has one feature, which is sending structure bounding boxes for MiniHUD.

What is your command rate in Litematica set to? If it's still at 64 commands per tick, try lowering it to for example 24 or 16 commands per tick. It's at Generic -> `commandLimitPerTick`.

**Reply by Leopotter1218** [reply]
it is at 6

**Reply by maruohon** [reply]
Can you send me the schematic you are testing with, and also the exact position in world (the placement origin, and any other settings like rotations) where you are pasting it, unless it's the same in all positions and rotations.

Also if you use Discord, it might be more convenient to figure this out if you hop on to my server: https://discord.gg/2FgywHj

**Reply by Leopotter1218** [reply]
yea sure due to academics i cant use discord till tmrw so will join and talk abt this day after tomorrow thank you so much for your active support till now
